Kernel/TermPartialOrdering.cpp

```diff
@@ -0,0 +1,366 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+/**
+ * @file TermPartialOrdering.cpp
+ * Implements class TermPartialOrdering.
+ */
+
+#include <cstring>
+#include <iomanip>
+
+#include "TermPartialOrdering.hpp"
+
+namespace Kernel {
+
+using namespace std;
+
+// Helper functions
+
+Result poCompToResult(PoComp c) {
+  switch (c) {
+    case PoComp::GREATER:
+      return Result::GREATER;
+    case PoComp::EQUAL:
+      return Result::EQUAL;
+    case PoComp::LESS:
+      return Result::LESS;
+    case PoComp::NGEQ:
+    case PoComp::NLEQ:
+    case PoComp::INCOMPARABLE:
+      return Result::INCOMPARABLE;
+    default:
+      ASSERTION_VIOLATION;
+  }
+}
+
+PoComp resultToPoComp(Result r, bool reversed) {
+  switch (r) {
+    case Result::GREATER:
+      return reversed ? PoComp::LESS : PoComp::GREATER;
+    case Result::EQUAL:
+      return PoComp::EQUAL;
+    case Result::LESS:
+      return reversed ? PoComp::GREATER : PoComp::LESS;
+    case Result::INCOMPARABLE:
+      return reversed ? PoComp::NLEQ : PoComp::NGEQ;
+    default:
+      ASSERTION_VIOLATION;
```

Comment:
@MichaelRawson , don't we have a new official "pattern" for how to deal with enums and "potentially unlisted cases" / warnings?

Reply:
I like the "assertion under `default` label" pattern more because it works regardless of what the function does outside of the `switch`, and there are several occurrences of this pattern in the code already. But I don't mind the other pattern either.

Reply:
A slight annoyance is that if the enum is extended at some point the `default` label masks the warning we'd otherwise get - but you make a good point about the rest of the function. Such is life.

Reply:
Ah, I forgot about this. Yes, then even if I don't expect the enum to be extended, let's have the other solution.